NumLib/TimeStepping/Algorithms/IterationNumberBasedTimeStepping.cpp

```diff
@@ -119,6 +126,35 @@ double IterationNumberBasedTimeStepping::getNextTimeStepSize() const
     return dt;
 }
 
+double IterationNumberBasedTimeStepping::checkSpecificTimeReached(const double h_new)
```

Comment:
Since only FixedTimeStepping dose not need ` _fixed_output_times` and `checkSpecificTimeReached`, we can move these to the base class `TimeStepAlgorithm` in future to avoid code duplication. For this PR, it is Ok.